include/common/BufferHelper.h

```diff
@@ -0,0 +1,16 @@
+#ifndef BUFFER_HELPER_H
+#define BUFFER_HELPER_H
+
+#include <cstdint>
+
+typedef struct AugmentedBuffer
+{
+    void* buffer_pointer;
```

Comment:
nit: This is maybe a larger topic around naming conventions, style guide, etc. but in C I very often see names like `ptr` or `sz` so maybe we can decide later if we want to stick with the more verbose Java style or the more terse C style

Reply:
Sure, I can do either but I highly prefer more verbose names.